### PR TITLE
[Bug] IE, Firefox, & Edge Layout Issues

### PR DESCRIPTION
**Pull request by grahamhency**
## PR Checklist
Please check if your PR fulfills the following requirements:
- [x] The commit message follows our guidelines: https://github.com/mobi/goponents/blob/master/CONTRIBUTING.md
~~Tests for the changes have been added (for bug fixes / features)~~
~~Docs have been added / updated (for bug fixes / features)~~

## PR Type
What kind of change does this PR introduce?
- [x] Bugfix
- [ ] Feature
- [x] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?
See my review comments, but there were various minor issues in regards to layout and styles across Firefox, Edge, and IE11.

## What is the new behavior?
Fixed those issues to the best of our capabilities.

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/go-lib/src/styles/_mixins.scss [hunk 1]
```diff
@@ -65,6 +65,10 @@ $z-index: (
   letter-spacing: .5pt;
   outline: none;
   width: 100%;
+
+  &::-ms-clear {
```
Comment (grahamhency) [inline projects/go-lib/src/styles/_mixins.scss]: Added this for the same reason on the `go-table` search.

File: projects/go-lib/src/styles/_reset.scss [hunk 2]
```diff
@@ -51,8 +51,4 @@ q:before, q:after {
 table {
   border-collapse: collapse;
   border-spacing: 0;
-}
-
-* {
-  box-sizing: border-box;
```
Comment (grahamhency) [inline projects/go-lib/src/styles/_reset.scss]: Moved this to `globals`

File: projects/go-lib/src/lib/components/go-select/go-select.component.html [hunk 3]
```diff
@@ -9,6 +9,7 @@
 
 <ng-select
   [appendTo]="appendTo"
+  [clearable]="clearable"
```
[the remark this answers is not in the document]
Comment (jaredami) [reply projects/go-lib/src/lib/components/go-select/go-select.component.html]: Think we should add this to the docs, or do as Eli suggested and add a link to the ng-select docs?

File: projects/go-lib/src/lib/components/go-header/go-header.component.html [hunk 4]
```diff
@@ -1,23 +1,30 @@
-<header class="go-header">
-  <div class="go-header__left"
-       [ngClass]="{ 'go-header__left--dark': brandColorIsDark }">
-    <div class="go-header__menu-container"
-         [ngStyle]="{ 'background': brandColor }"
-         (click)="toggleSideMenu()">
+<header
+  class="go-header">
```
[the remark this answers is not in the document]
Comment (grahamhency) [reply projects/go-lib/src/lib/components/go-header/go-header.component.html]: Yeah, I moved it back up to the first line.

File: projects/go-lib/src/lib/components/go-select/go-select.component.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (grahamhency) [reply projects/go-lib/src/lib/components/go-select/go-select.component.html]: Yeah, I added docs for it. Good catch.

**Pull request merged**